index.js

```diff
@@ -620,21 +639,20 @@
     );
   };
 
-  //  equalTypes :: (Type, Type) -> Boolean
-  var equalTypes = function equalTypes(t1, t2) {
+  //  equalTypes :: (Type, Type, Boolean) -> Boolean
+  var equalTypes = function equalTypes(t1, t2, loose) {
+    if (t1.type === 'INCONSISTENT' || t2.type === 'INCONSISTENT') return loose;
```

Comment:
We set `loose` to `true` when generating error messages, which allows us to display `['foo', false] :: Array ???` rather than `['foo', false] :: ???`.